var/spack/repos/builtin/packages/texinfo/package.py

```diff
@@ -34,6 +34,7 @@ class Texinfo(AutotoolsPackage, GNUMirrorPackage):
     version("5.0", sha256="2c579345a39a2a0bb4b8c28533f0b61356504a202da6a25d17d4d866af7f5803")
 
     depends_on("perl")
+    depends_on("gettext@0.20.1:", type="build", when="@6.7.0:")
```

Comment:
These versions are based on https://lists.gnu.org/archive/html/info-gnu/2019-09/msg00007.html but to me that doesn't necessarily imply that 6.7 was the first version to depend on gettext or that 0.20.1 is the minimum version. Did anyone actually check these to see if 6.6 uses gettext or if older gettext works?

Reply:
Good catch.  here are the details:

```
> pcregrep '^\d\.\d|gettext' NEWS
7.0 (7 November 2022)
6.8 (3 July 2021)
  . autoconf 2.71, automake 1.16.3, gettext 0.21
6.7 (23 September 2019)
  . gettext 0.20.1, automake 1.16.1
6.6 (16 February 2019)
6.5 (12 September 2017)
6.4 (23 June 2017)
  . autoconf 2.69, automake 1.15, gettext 0.19.8
6.3 (10 September 2016)
6.2 (withdrawn)
6.1 (06 February 2016)
  . automake-1.15, gettext-0.19.6.
6.0 (26 June 2015)
  . automake-1.15, gettext-0.19.4.
5.2 (26 September 2013)
* Distribution: automake-1.14, gettext-0.18.3.1.
5.1 (12 March 2013)
5.0 (16 February 2013)
  . autoconf 2.69, automake 1.13.1, gettext 0.18.2.
```

Reply:
Yep, I would personally always add the dep (you can quickly build the old version we ship to see) and remove the minimum version constraint, so go back to the way you had it.

Reply:
ok this is fun:

```console
> pcregrep '^\d\.\d|gettext' NEWS | grep -B 1 gettext | grep -v -- -- | pcregrep -A 3 '^\d\.\d' | pcregrep -o '(^[4-7]\.\d+|gettext[\- ]+\d+\.\d+(\.\d+)?)' | perl -pe 's/\ngettext//gm' | paste -s -d' \n'  | perl -pe 's/(.*)\s+gettext[\s-]+(.*)/depends_on("gettext@\1:", type="build", when="@\2:")/'
depends_on("gettext@6.8:", type="build", when="@0.21:")
depends_on("gettext@6.7:", type="build", when="@0.20.1:")
depends_on("gettext@6.4:", type="build", when="@0.19.8:")
depends_on("gettext@6.1:", type="build", when="@0.19.6:")
depends_on("gettext@6.0:", type="build", when="@0.19.4:")
depends_on("gettext@5.2:", type="build", when="@0.18.3:")
depends_on("gettext@5.0:", type="build", when="@0.18.2:")
depends_on("gettext@4.12:", type="build", when="@0.17:")
depends_on("gettext@4.11:", type="build", when="@0.16.1:")
depends_on("gettext@4.7:", type="build", when="@0.14.1:")
depends_on("gettext@4.6:", type="build", when="@0.12.1:")
depends_on("gettext@4.3:", type="build", when="@0.11.5:")
depends_on("gettext@4.2:", type="build", when="@0.11.1:")
depends_on("gettext@4.1:", type="build", when="@0.11:")
```